src/Bicep.Cli/Services/CompilationService.cs

```diff
@@ -94,6 +97,24 @@ public async Task<Compilation> CompileAsync(string inputPath, bool skipRestore)
             return compilation;
         }
 
+
+        public ProgramSyntax CompileParams(string inputPath, bool skipRestore)
+        {
+            var fileText = File.ReadAllText(inputPath);
```

Comment:
Use `fileResolver.TryRead(...)` here. You'll need `PathHelper.FilePathToFileUrl(inputPath)` to get the file path in `Uri` format.